controllers/watcher_common.go

```diff
@@ -24,22 +25,35 @@ import (
 )
 
 const (
-	passwordSecretField   = ".spec.secret"
-	prometheusSecretField = ".spec.prometheusSecret"
+	passwordSecretField     = ".spec.secret"
+	prometheusSecretField   = ".spec.prometheusSecret"
+	caBundleSecretNameField = ".spec.tls.caBundleSecretName"
+	tlsAPIInternalField     = ".spec.tls.api.internal.secretName"
+	tlsAPIPublicField       = ".spec.tls.api.public.secretName"
 )
 
 var (
 	apiWatchFields = []string{
 		passwordSecretField,
 		prometheusSecretField,
+		tlsAPIInternalField,
+		tlsAPIPublicField,
+		caBundleSecretNameField,
```

Comment:
Yes, we were missing this. I guess we need it for the case of certificate rotation or something like that.